src/frontend/src/pages/ProjectPage/components/EditContractWizard/components/BasePositionPicker.tsx

```diff
@@ -16,40 +17,18 @@ const BasePositionPicker: React.FC<BasePositionPickerProps> = ({
     selectedBasePositionId,
     onSelect,
 }) => {
-    const apiClients = useApiClients();
-    const telemetryLogger = useTelemetryLogger();
-
-    const [basePositions, setBasePositions] = React.useState<BasePosition[]>([]);
-    const [isFetchingBasePositions, setIsFetchingBasePositions] = React.useState(false);
-    const [basePositionsError, setBasePositionsError] = React.useState<Error | null>(null);
-    const fetchBasePositions = async () => {
-        setIsFetchingBasePositions(true);
-        setBasePositionsError(null);
-
-        try {
-            const response = await apiClients.org.getAsync<BasePosition[]>(
-                combineUrls('positions', "basepositions?$filter=projectType eq 'PRD-Contracts'")
-            );
-            setBasePositions(response.data);
-        } catch (e) {
-            telemetryLogger.trackException(e);
-            setBasePositionsError(e);
-        }
-
-        setIsFetchingBasePositions(false);
-    };
-
-    React.useEffect(() => {
-        fetchBasePositions();
-    }, []);
+    const { basePositions, basePositionsError, isFetchingBasePositions } = useAppContext().useBasePositions;
```

Comment:
Should use the useBasePositions hook directly here, instead of providing the result of the hook through the app context